tests/test_compareSummary.cpp

```diff
@@ -0,0 +1,406 @@
+/*
+   Copyright (C) 2011  Statoil ASA, Norway.
+   The file 'view_summary.c' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+// This script is based on/inspired by  view_summay.c found on  
+
+#include <stdlib.h>
+#include <string.h>
+#include <signal.h>
+#include <vector> 
+#include <iostream>
+#include <algorithm>
+
+#include <ert/util/ert_api_config.h>
+
+#ifdef ERT_HAVE_GETOPT
+#include <getopt.h>
+#endif
+
+#include <ert/util/util.h>
+#include <ert/util/stringlist.h>
+
+#include <ert/ecl/ecl_kw.h>
+#include <ert/ecl/ecl_sum.h>
+
+#include <stdbool.h>
+#include <math.h>
+#include <time.h>
+#include <locale.h>
+
+
+#include <ert/util/hash.h>
+#include <ert/util/set.h>
+#include <ert/util/util.h>
+#include <ert/util/vector.h>
+#include <ert/util/int_vector.h>
+#include <ert/util/bool_vector.h>
+#include <ert/util/time_t_vector.h>
+#include <ert/util/time_interval.h>
+
+#include <ert/ecl/ecl_util.h>
+#include <ert/ecl/ecl_smspec.h>
+#include <ert/ecl/ecl_sum_data.h>
+#include <ert/ecl/smspec_node.h>
+
+#include <opm/common/ErrorMacros.hpp>
+
+
+#define DATE_HEADER         "-- Days   dd/mm/yyyy   "
+#define DATE_STRING_LENGTH 128
+
+  struct ecl_sum_struct {
```

Comment:
No; this can not be merged. 

It is not entirely clear to me what you wanted to achieve this way, but if there are shortcoming/problems with the `ecl_sum` implementation in ert we should fix that - I am happy to assist. Copying in parts of the library implementation like this is not a viable solution.